clone/clone.go

```diff
@@ -333,14 +320,24 @@ Specific cloning functions begin here.
 
 // GoldenGate simulates a GoldenGate cloning reaction. As of right now, we only
 // support BsaI, BbsI, BtgZI, and BsmBI.
-func GoldenGate(sequences []Part, enzymeStr string) ([]string, []string, error) {
+func (enzymeManager EnzymeManager) GoldenGate(sequences []Part, enzymeStr string) (openConstructs []string, infiniteLoops []string, err error) {
```

Comment:
should GoldenGate really be a method of enzymeManager rather than its function?

Reply:
After learning more about how Golden Gate works IRL and a little bit more about SynBio- probably not. 

It seems more appropriate for Golden Gate to just accept something like a 'cuttingEnzyme' of type 'Enzyme' and have the caller lookup whatever enzyme they want.

This is also nice because now Golden Gate doesn't have to return an error anymore.

Reply:
I renamed it to 'ExampleEnzymeManager_GoldenGate' for the linter- it's still there!

Reply:
I personally would like the change of GoldenGate to be its own function, independent of an enzymeManager.

Reply:
Updated